### PR TITLE
Add gRPC error handling doc

### PR DESCRIPTION
**Pull request by JamesNK**
Docs about gRPC error handling and the new rich error handing.

Ready for review, **not ready for merge** (waiting for error handling to be released).

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/error-handling.md](https://github.com/dotnet/AspNetCore.Docs/blob/901de106e1fbe59b804875fc66aba58815af1d5c/aspnetcore/grpc/error-handling.md) | [aspnetcore/grpc/error-handling](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/error-handling?branch=pr-en-us-31386) |

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/error-handling.md](https://github.com/dotnet/AspNetCore.Docs/blob/63dd44c8b3bcef1775ab86b71492e90d692dea27/aspnetcore/grpc/error-handling.md) | [aspnetcore/grpc/error-handling](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/error-handling?branch=pr-en-us-31386) |

### CODE REVIEW

**Comment by JamesNK** [comment]
cc @tonydnewell

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/grpc/error-handling.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        if (string.IsNullOrEmpty(request.Name))
+        {
+            throw new RpcException(new Status(StatusCode.InvalidArgument, "Name is required."));
+        }
+        return Task.FromResult(new HelloReply { Message = $"Hello {request.Name}" });
+    }
+}
+```
+
+Thrown exception types that aren't `RpcException` also cause the call to fail, but with a an `UNKNOWN` status code and a generic message `Exception was thrown by handler`.
```
Comment (wadepickett) [inline aspnetcore/grpc/error-handling.md:81]:
```suggestion
Thrown exception types that aren't `RpcException` also cause the call to fail, but with an `UNKNOWN` status code and a generic message `Exception was thrown by handler`.
```

**Review by wadepickett — APPROVED** [review]
Approved with a minor suggestion.  Looks great, thanks James.

I can help by moving the inline code to an app sample if you like. Perhaps right after publishing so I don't complicate this PR.

**Comment by Rick-Anderson** [comment]
@tonydnewell can you take a quick look?

**Comment by tonydnewell** [comment]
@Rick-Anderson looks ok

**Pull request merged**